lib/parsers.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileoverview Defines a storage for parsers.
+ * @author Ives van Hoorne
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+class Parsers {
+    constructor() {
+        this._parsers = Object.create(null);
+    }
+
+    /**
+     * Registers a custom parser in storage
+     * @param {string} parserId Parser id.
+     * @param {Function} parserModule Parser handler.
+     * @returns {void}
+     */
+    define(parserId, parserModule) {
+        this._parsers[parserId] = parserModule;
+    }
+
+    /**
+     * Access parser handler by id.
+     * @param {string} parserId Parser id.
+     * @returns {void}
+     */
+    get(parserId) {
+        const parser = this._parsers[parserId];
+
+        if (!parser) {
+            return require(parserId);
```

Comment:
Should `require(parserId)` be cached into the parser map in this case?

Reply:
Hmm, node already caches requires by default (https://nodejs.org/api/modules.html#modules_require_cache), but I could add it for explicitness.